packages/lexicon/src/util.ts

```diff
@@ -145,3 +145,18 @@ export function requiredPropertiesRefinement<
     }
   }
 }
+
+// to avoid bugs like #0189 this needs to handle both
+// explicit and implicit #main
+const refsContainType = (refs: string[], type: string) => {
+  const lexUri = toLexUri(type)
+  if (refs.includes(lexUri)) {
+    return true
+  }
+
+  if (lexUri.endsWith('#main')) {
+    return refs.includes(lexUri.replace('#main', ''))
```

Comment:
This would also accept strings like `com.example#main#main` and simplify to `com.example`

Reply:
Yep, I took that into consideration, but AFAIK `com.example#main#main` is not a valid type

Reply:
I would be inclined to use something like `lexUri.slice(0, -"#main".length)` instead

Reply:
that check could also be moved up to `toLexUri`

Reply:
there should never be more than one hash fragment in a lex uri

something like `if (str.split('#').length > 2) throw new Error()`

Reply:
As @dholms suggested I've added a heck and tests to `toLexUri`